### ISSUE TITLE
Can't get new-haddock to organize docs for lib and exe on a single page

### ISSUE THREAD

**Issue by Mikolaj** [haskell/cabal#4873]
Edit: I was wrong, I missed the message with a separate address for the lib docs earlier on in the log. See also the next comment.

~When I specify `--haddock-executables` I only get exe docs and there's no way to also include lib docs. Even with `--haddock-all` I only get the exe docs. I can get lib docs if I provide no flags, but then there are no exe docs. Version 2.0.~

**Comment by Mikolaj** [comment]
Actually, I now see both lib and exe docs are generated, but they are in separate directories and I expected them to be all visible in one documentation hierarchy. If that is intended, please close the issue. However, if so, is there some auto-generated page with links to docs of all package components? That would be useful. And ideally, I'd have all in one hierarchy, without rendering the description from .cabal file multiple times. Then I could even try and upload such documentation to Hackage.

**Comment by Mikolaj** [comment]
I think, this is now closely related to handling of multiple libraries in a single cabal file and at this point the problem resides mainly or solely with haddock, not cabal. Closing (a ticket was recently opened for that at haskell/haddock#1363). Once multiple libraries are fixed and there is still interest (I no longer need this, since I worked around making an internal library from the modules related to the executable; alas, haddocks for multiple libraries don't work yet), this can be re-opened in the haddock repo.
